include/envoy/event/dispatcher.h

```diff
@@ -77,6 +78,21 @@ class DispatcherBase {
    */
   virtual Event::TimerPtr createTimer(TimerCb cb) PURE;
 
+  /**
+   * Allocates a scaled timer. @see Timer for docs on how to use the timer.
+   * @param timer_type the type of timer to create.
+   * @param cb supplies the callback to invoke when the timer fires.
+   */
+  virtual Event::TimerPtr createScaledTimer(Event::ScaledRangeTimerManager::TimerType timer_type,
```

Comment:
Would it make sense to move `TimerType` somewhere else so that `Dispatcher` doesn't need to depend on `ScaledRangeTimerManager`?

Reply:
Yeah I'm not attached to the placement, and I don't like how verbose the references are to it (`Event::ScaledRangeTimerManager::TimerType`). What about `::Envoy::Event::ScaledTimerType`?

Reply:
Given that callers are expected to use this on the `Dispatcher` function at the top-level, I'd say move the enum into `Dispatcher`, or just into the `Envoy::Event` namespace.

Reply:
Put it in `Envoy::Event`.